### PR TITLE
Add requested CPU and memory for NodeSample

### PR DESCRIPTION
**Pull request by adriaferre**

### CODE REVIEW

File: src/kubelet/group.go [hunk 1]
```diff
@@ -66,12 +66,26 @@ func (r *kubelet) Group(definition.SpecGroups) (definition.RawGroups, *data.Erro
 			Errors:      []error{fmt.Errorf("error querying ApiServer: %v", err)},
 		}
 	}
+
+	podsResource, err := metric.GetMetricsPodsData(r.client)
+	var nodeRequestedCPU, nodeRequestedMemory int64 = 0, 0
+	if err != nil {
```
Comment (ahoulgrave) [inline src/kubelet/group.go:72]: maybe allow to fail, but do log the error so we can debug

Comment (ahoulgrave) [reply src/kubelet/group.go:72]: if we allow the request to fail, we may collect misleading data

File: src/kubelet/metric/metric.go [hunk 2]
```diff
@@ -43,6 +48,28 @@ func GetMetricsData(c client.HTTPClient) (v1.Summary, error) {
 
 }
 
+// GetMetricsPodsData calls kubelet /pods endpoint and returns unmarshalled response
```
Comment (alejandrodnm) [inline src/kubelet/metric/metric.go:51]: The integration at some point already queries the pods endpoint of the kubelet using this https://github.com/newrelic/nri-kubernetes/blob/main/src/kubelet/metric/pods.go#L42 I think what you should do, is hook your code at some point in the integration execution, where the integration already has the data you want.

We want to limit the amount of requests we make from the integration, to reduce as much as possible the footprint of the agent on the Kubelet and apiserver.

I think at this point https://github.com/newrelic/nri-kubernetes/blob/main/src/kubelet/group.go#L69 the `rawGroups` object already has the data you want, you'll just have to traverse the object, aggregate the data you want and add the data you want the same way the labels and allocatable resources are being added.

The team that maintains this project is super busy right now, but I can lend a hand and review your PR, ping me whenever you want.

**Review by alejandrodnm — CHANGES_REQUESTED** [review]

File: src/kubelet/metric/metric.go (hunk 2 above)
Comment (ahoulgrave) [reply src/kubelet/metric/metric.go:51]: At this point: https://github.com/newrelic/nri-kubernetes/blob/main/src/kubelet/group.go#L69

The information i have on pods is:
```
pod: map[
    createdAt:2020-12-23 09:46:15 +0000 UTC 
    createdBy:nginx-deployment-695987c984 
    createdKind:ReplicaSet 
    deploymentName:nginx-deployment 
    isReady:True 
    isScheduled:True 
    labels:map[
        app:nginx pod-template-hash:695987c984
    ]
    namespace:default 
    nodeIP:192.168.49.2 
    nodeName:minikube 
    podName:nginx-deployment-695987c984-t229l 
    startTime:2020-12-23 09:46:15 +0000 UTC 
    status:Running
]
```

where

```
-> % k get pod nginx-deployment-695987c984-t229l -o jsonpath="{.spec.containers[0].resources}" |jq
{
  "limits": {
    "cpu": "500m",
    "memory": "128Mi"
  },
  "requests": {
    "cpu": "250m",
    "memory": "64Mi"
  }
}
```

we'll need to find a place where this data is available

**Comment by ahoulgrave** [comment]
Opened a new PR : https://github.com/newrelic/nri-kubernetes/pull/61

File: src/kubelet/metric/metric.go (hunk 2 above)
Comment (ahoulgrave) [reply src/kubelet/metric/metric.go:51]: NVM: it's in the containers, not the pods :) 

```
ncontainer: map[
    containerID:33a266b2b09f33ac22aaafe5b3bedf0cbee14ee24b3cb73c4698ddb3448afee2
    containerImage:nginx:1.14.2
    containerImageID:nginx@sha256:f7988fb6c02e0ce69257d9bd9cf37ae20a60f1df7563c3a2a6abe24160306b8d
    containerName:nginx 
    container_cpu_cfs_periods_total:3 
    container_cpu_cfs_throttled_periods_total:0 
    container_cpu_cfs_throttled_seconds_total:0 
    container_memory_mapped_file:0 
    cpuLimitCores:500 
    cpuRequestedCores:250 
    deploymentName:nginx-deployment 
    isReady:true 
    labels:map[
        app:nginx 
        pod-template-hash:695987c984
    ] 
    memoryLimitBytes:134217728 
    memoryRequestedBytes:67108864 
    namespace:default 
    nodeIP:192.168.49.2 
    nodeName:minikube 
    podName:nginx-deployment-695987c984-t229l 
    restartCount:0 
    startedAt:2020-12-23 09:46:28 +0000 UTC 
    status:Running
]
```

**Pull request closed without merging**